lm_eval/models/vllm_causallms.py

```diff
@@ -182,13 +175,24 @@ def _model_generate(
                 temperature=0, prompt_logprobs=1, max_tokens=1
             )
         if self.data_parallel_size > 1:
+            # vLLM hangs if tensor_parallel > 1 and resources are set in ray.remote
+            # also seems to only work with decorator and not with ray.remote() fn
+            # see https://github.com/vllm-project/vllm/issues/973
+            @ray.remote
```

Comment:
Just confirming this is compatible with older vLLM versions?

Reply:
It's not even working for the latest version they merged today! It worked fine up until `0.3.2`. Added an assertion for it to be < 0.3.3 if DP. Let me know if you want it to be more restrictive. I tried checking on `0.2.5` as well, but the dependencies are a mess and haven't been able to vLLM to run.

Reply:
I think I can just get DP to work. But not TP _and_ DP together. Problem seems to be theres no way to set device(s) when initialising.

Reply:
Hm, yeah let's pin to `0.3.2` then for now!

Reply:
ok done! also fixed the typos.

Reply:
Thanks a ton!